tests/pact/fqns.repl

```diff
@@ -77,3 +77,165 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+; In the following tests, we define a module `test-mod-redeploy-ref`, and then
+; redeploy the same module with the change to one capability: `test`.
+; In the old version, the `test` capability fails, in the new one it passes.
+
+(begin-tx)
+; First, demonstrate the behavior prior to pact-4.8.
+(env-exec-config ["DisablePact48"])
+
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+; Before pact-4.8, the updated capability will be ignored, and calls to a function
+; requiring that capability will fail. This is true whether we invoke `f` with or
+; without its module name, and with or without its namespace.
+(expect-failure "Execution expected to fail due to assertion failure \
```

Comment:
These are now too verbose, so as long as we're bikeshedding on this I'll continue to be pedantic:

"Execution expected to fail" -- implied by `expect-failure`, remove.

"due to assertion failure within defcap invoke" -- the real reason we're doing this is just to test dispatch over various upgrades, and failure is the chosen mechanism to reveal defcap acquisition (since they are side-effect only, otherwise we'd have to write a row to the db etc). Finally, in this pass we're just demonstrating pre-upgrade behavior. I would simply say something like "Verify defcap resolution" as this succinctly documents _intent_.

"invoking with unqualified reference" -- this is fine but a little verbose, and it's nice if expectation docs can fit on a single line. Also if the above comment is true, you won't need to document invocation as it's not relevant.

My suggestion would therefore be "Demonstrate defcap resolution" since this is not expected to be broken at this point, and to distinguish from later use cases, and assuming that we don't need the expectation triple.